modules/programs/gitui/gitui.nix

```diff
@@ -0,0 +1,74 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.gitui;
+
+  yamlFormat = pkgs.formats.yaml { };
+
+  inherit (pkgs.stdenv.hostPlatform) isDarwin;
+
+in {
+  meta.maintainers = [ maintainers.mifom ];
+
+  options.programs.gitui = {
+    enable =
+      mkEnableOption "gitui, blazing fast terminal-ui for git written in rust";
+
+    key_config = mkOption {
+      type = types.str;
+      default = "";
+      example = ''
+        exit: ( code: Char('c'), modifiers: ( bits: 2,),),
+        quit: ( code: Char('q'), modifiers: ( bits: 0,),),
+        exit_popup: ( code: Esc, modifiers: ( bits: 0,),),
+      '';
+      description = ''
+        Key config in Ron file format. It written to
+        <filename>~/.config/gitui/key_config.ron</filename>.
+      '';
+    };
+    theme = mkOption {
+      type = types.str;
+      default = ''
+        (
+          selected_tab: Reset,
+          command_fg: White,
+          selection_bg: Blue,
+          cmdbar_extra_lines_bg: Blue,
+          disabled_fg: DarkGray,
+          diff_line_add: Green,
+          diff_line_delete: Red,
+          diff_file_added: LightGreen,
+          diff_file_removed: LightRed,
+          diff_file_moved: LightMagenta,
+          diff_file_modified: Yellow,
+          commit_hash: Magenta,
+          commit_time: LightCyan,
+          commit_author: Green,
+          danger_fg: Red,
+          push_gauge_bg: Blue,
+          push_gauge_fg: Reset,
+        )
+      '';
+      description = ''
+        Theme in Ron file format. It written to
+        <filename>~/.config/gitui/theme.ron</filename>.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ pkgs.gitui ];
+
+    xdg.configFile."gitui/theme.ron" = { text = cfg.theme; };
+    xdg.configFile."gitui/key_config.ron" = {
+      text = if cfg.key_config != "" then
+        cfg.key_config
+      else
+        (builtins.readFile ./default_key_config.ron);
```

Comment:
Could you also apply the logic about the `cfg.theme` above?

Also, does the default key configuration need to be bundled with the HM module?
If it needs to be bundled, then could you update the `keyConfig` option with:
``` nix
mkOption {
  # other option attributes
  default = ./default_key_config.ron;
  defaultText = literalExpression ("''" + (builtins.readFile ./default_key_config.ron) + "''");
}
```

Reply:
This needs revisiting. Is it absolutely necessary to provide the entire default keybinding and theme files here? Are the files here matching the actual defaults of gitui or do they differ? If they are the same, would it be possible to retrieve the files from the actual package used?

Also, how does this affect the HM manual, it seems like this would add a lot of lines to it?

Reply:
There are updates about key config in version 0.19, so now it's not nesessary.

I've updated this in https://github.com/nix-community/home-manager/pull/2446/commits/3b1d6e4668a92d3b700810a0f39b08f388ad81c8